torchbenchmark/util/framework/huggingface/model_factory.py

```diff
@@ -146,3 +159,43 @@ def eval(self) -> Tuple[torch.Tensor]:
         else:
             return (out["logits"], )
 
+
+class HuggingFaceGenerationModel(HuggingFaceModel):
+    task = NLP.GENERATION
+    DEFAULT_EVAL_BSIZE = 1
+
+    """
+    Instead of just running __call__ on the model, use generate to generate
+    text.
+    """
+    def __init__(self, name, test, device, jit=False, batch_size=None, extra_args=[]):
+        super().__init__(name=name, test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+        # Make this configurable with extra_args
+        # NB: this is *fixed* generation size as eos_token_id is None
+        # These params were cribbed off of
+        # https://github.com/younesbelkada/hf-torch-compile-benchmark
+        generation_config = GenerationConfig(
+            max_new_tokens=256,
+            pad_token_id=0,
+            eos_token_id=None,
+            do_sample=False,
+            num_beams=1,
+            use_cache=True,
+        )
+        self.model = GenerationWrapper(self.model, generation_config)
+
```

Comment:
It seems to me we should add `def train(self): raise NotImplementError()` here, because the old training loop wouldn't make sense with Generation model.

Reply:
yes, will add